sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -195,6 +195,12 @@ object SQLExecution extends Logging {
                   }
                 }
               }
+
+              if (executionId == rootExecutionId) {
```

Comment:
let's leave some code comments here to explain it.

It will be better if we can cancel uesless jobs earlier when nested query executions are completed, but we can do it in a followup.

Reply:
@cloud-fan Yes. Added the comments. Please help to review again. Thanks.